src/unix/mod.rs

```diff
@@ -621,6 +620,7 @@ extern {
                     addrlen: *mut socklen_t) -> ::ssize_t;
     pub fn mkfifo(path: *const c_char, mode: mode_t) -> ::c_int;
 
+    #[cfg(not(target_os = "sunos"))]
```

Comment:
Could this function definition move directly into `bsd/mod.rs` and `notbsd/mod.rs`? We tend to avoid `#[cfg(not)]` if possible in libc currently